File: R/luhn.R
luhnCheck <- function(number){
    if(nchar(number) <= 1) return(FALSE)

    digits <- unlist(strsplit(gsub(" ", "", number), ""))

    digits <- suppressWarnings(as.numeric(digits))
    if(any(is.na(digits))) return(FALSE)

    indexSelect <- seq(length(digits) - 1, 1, -2)

    digits[indexSelect] <- digits[indexSelect]*2
    digits[digits > 9] <- digits[digits > 9] - 9
    return(sum(digits))
}

luhnCheck("4539 3195 0343 6467") # %% 10 == 0

luhnCheck("8273 1232 7352 0569") # %% 10 == 0

luhnCheck("A4230B")

File: python/luhn.py
def luhnCheck(n):
    if(len(n) <= 1): return(False)
    n = n.replace(" ", "") # Removing spaces
    digits = [int(x) for x in n]
    indexSelect = list(range(0, len(digits), 2))

    digits = [x*2 if x.index % 2 == 0 else x for x in digits ]
    digits = [x - 9 if x > 9 else xfor x in digits if x > 9]
    print(digits)
    return(sum(digits))

teste = [1,2, 3, 4]
teste[0:len(teste):2] = teste[0:len(teste):2]*

luhnCheck("4539 3195 0343 6467")

teste = "4539 3195 0343 6467"
list[teste.replace(" ", "")]


